# Exploration des données

Objectif : comprendre la structure, la qualité et les limites des données
(population et revenus) avant tout traitement ou analyse.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
base_path = "/content/drive/MyDrive/Projet Data"


In [ ]:
import os

project_name = "projet_1_diagnostic_territorial"
project_path = os.path.join(base_path, project_name)

folders = [
    "data/raw",
    "data/clean",
    "notebooks",
    "dashboard",
    "outputs"
]

for folder in folders:
    os.makedirs(os.path.join(project_path, folder), exist_ok=True)

project_path


'/content/drive/MyDrive/Projet Data/projet_1_diagnostic_territorial'

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
from scipy import stats

In [ ]:

pop = pd.read_excel('/content/drive/MyDrive/Projet Data/projet_1_diagnostic_territorial/data/raw/POPULATION_MUNICIPALE_COMMUNES_FRANCE.xlsx')
rev = pd.read_csv('/content/drive/MyDrive/Projet Data/projet_1_diagnostic_territorial/data/raw/revenu-des-francais-a-la-commune-1765372688826.csv', sep= ';')

pop.head()


/tmp/ipython-input-260840590.py:2: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  rev = pd.read_csv('/content/drive/MyDrive/Projet Data/projet_1_diagnostic_territorial/data/raw/revenu-des-francais-a-la-commune-1765372688826.csv', sep= ';')


objectid  reg dep    cv codgeo                 libgeo  p13_pop  p14_pop  \
0    115658   52  85  8502  85062            Châteauneuf    968.0    993.0   
1    115659   26  58  5808  58300                   Urzy   1839.0   1835.0   
2    115660   43  70  7012  70137  Chassey-lès-Montbozon    218.0    217.0   
3    115661   21  51  5123  51649      Vitry-le-François  13174.0  13144.0   
4    115662   11  78  7811  78638         Vaux-sur-Seine   4749.0   4715.0   

   p15_pop  p16_pop  p17_pop  p18_pop  p19_pop  p20_pop  p21_pop  
0   1013.0   1027.0     1056   1085.0   1114.0   1118.0   1134.0  
1   1828.0   1802.0     1775   1749.0   1746.0   1747.0   1742.0  
2    216.0    215.0      217    215.0    215.0    220.0    225.0  
3  12805.0  12552.0    12133  11743.0  11376.0  11458.0  11454.0  
4   4788.0   4857.0     4927   4929.0   5010.0   5020.0   5083.0

In [ ]:
pop.shape

(34995, 15)

In [ ]:
pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34995 entries, 0 to 34994
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   objectid  34995 non-null  int64  
 1   reg       34995 non-null  int64  
 2   dep       34995 non-null  object 
 3   cv        34995 non-null  object 
 4   codgeo    34995 non-null  object 
 5   libgeo    34995 non-null  object 
 6   p13_pop   34994 non-null  float64
 7   p14_pop   34994 non-null  float64
 8   p15_pop   34994 non-null  float64
 9   p16_pop   34994 non-null  float64
 10  p17_pop   34995 non-null  int64  
 11  p18_pop   34992 non-null  float64
 12  p19_pop   34988 non-null  float64
 13  p20_pop   34978 non-null  float64
 14  p21_pop   34969 non-null  float64
dtypes: float64(8), int64(3), object(4)
memory usage: 4.0+ MB


Les types de données de la dataset population sont cohérentes

In [ ]:
pop.describe()

objectid           reg        p13_pop        p14_pop  \
count   34995.000000  34995.000000   34994.000000   34994.000000   
mean   127666.000000     49.700471    1846.004887    1868.220095   
std     10102.330672     25.592607    8415.276974    8478.899950   
min    110169.000000      1.000000       0.000000       0.000000   
25%    118917.500000     25.000000     199.000000     199.000000   
50%    127666.000000     43.000000     449.000000     452.000000   
75%    136414.500000     73.000000    1120.000000    1136.000000   
max    145163.000000     94.000000  458298.000000  466297.000000   

             p15_pop        p16_pop        p17_pop        p18_pop  \
count   34994.000000   34994.000000   34995.000000   34992.000000   
mean     1884.148111    1889.817912    1900.966967    1907.025206   
std      8534.926272    8551.638978    8583.400244    8627.491811   
min         0.000000       0.000000       0.000000       0.000000   
25%       199.000000     199.000000     199.000000     198.000000   
50%       454.000000     455.000000     457.000000     457.000000   
75%      1147.000000    1152.000000    1159.000000    1160.000000   
max    471941.000000  475438.000000  479553.000000  486828.000000   

             p19_pop        p20_pop        p21_pop  
count   34988.000000   34978.000000   34969.000000  
mean     1914.505773    1920.021242    1927.597701  
std      8678.183247    8698.041355    8737.679677  
min         0.000000       0.000000       0.000000  
25%       198.000000     197.000000     197.000000  
50%       457.000000     458.000000     458.000000  
75%      1162.000000    1164.000000    1167.000000  
max    493465.000000  498003.000000  504078.000000

On peut remarquer que la médianne de la population en 2021 par région est de 458, ce qui est très éloigné de la mooyenne avec un écart type de 8738, nous pouvons aussi remarquer des anomalies sur les valeurs extrêmes comme le minimum de 0 sur une ou plusieurs régions de la population enregistré

In [ ]:
pop.duplicated().sum() #Vérifier s'il y a des doublons dans la dataframe

np.int64(0)

La dataset ne présente pas de doublons

In [ ]:
pop.isna().sum().sort_values(ascending=False).head(10) #Trouver les valeurs manquantes de chaque colonnes

p21_pop    26
p20_pop    17
p19_pop     7
p18_pop     3
p13_pop     1
p16_pop     1
p15_pop     1
p14_pop     1
codgeo      0
libgeo      0
dtype: int64

La colonne de la population de 2021 présente des valeurs manquantes, mais comme il ne présente à peine 1% donc l'analyse n'en sera pas affecté

Le dataframe de la population a été explorée et vérifiée et il prêt pour le nettoyage, maintenant nous pouvons passer à celle des revenues.

In [ ]:
rev.head(10)

Nom géographique GMS Code géographique     Libellé géographique  \
0  L'Abergement-Clémenciat             01001  L'Abergement-Clémenciat   
1    L'Abergement-de-Varey             01002    L'Abergement-de-Varey   
2        Ambérieu-en-Bugey             01004        Ambérieu-en-Bugey   
3      Ambérieux-en-Dombes             01005      Ambérieux-en-Dombes   
4                  Ambléon             01006                  Ambléon   
5                 Ambronay             01007                 Ambronay   
6                 Ambutrix             01008                 Ambutrix   
7         Andert-et-Condon             01009             Aert-et-Coon   
8                Anglefort             01010                Anglefort   
9                 Apremont             01011                 Apremont   

   [DISP] Nbre de ménages fiscaux  \
0                           346.0   
1                           115.0   
2                          6855.0   
3                           800.0   
4                            51.0   
5                          1184.0   
6                           326.0   
7                           157.0   
8                           481.0   
9                           161.0   

   [DISP] Nbre de personnes dans les ménages fiscaux  \
0                                              895.0   
1                                              266.0   
2                                            15092.0   
3                                             2028.0   
4                                              107.0   
5                                             2874.0   
6                                              753.0   
7                                              380.0   
8                                             1122.0   
9                                              405.0   

   [DISP] Nbre d'unités de consommation dans les ménages fiscaux  \
0                                              590.8               
1                                              181.0               
2                                            10398.2               
3                                             1329.7               
4                                               76.6               
5                                             1915.7               
6                                              516.3               
7                                              256.8               
8                                              760.1               
9                                              265.9               

   [DISP] 1ᵉʳ quartile (€)  [DISP] Médiane (€)  [DISP] 3ᵉ quartile (€)  \
0                      NaN             25820.0                     NaN   
1                      NaN             24480.0                     NaN   
2                  15800.0             21660.0                 28430.0   
3                  20010.0             24610.0                 31180.0   
4                      NaN             24210.0                     NaN   
5                  19490.0             24860.0                 31530.0   
6                      NaN             26300.0                     NaN   
7                      NaN             24810.0                     NaN   
8                      NaN             24700.0                     NaN   
9                      NaN             24530.0                     NaN   

   [DISP] Écart interquartile (€)  ...  [DEC] 9ᵉ décile (€)  \
0                             NaN  ...                  NaN   
1                             NaN  ...                  NaN   
2                         12630.0  ...              39380.0   
3                         11170.0  ...              43610.0   
4                             NaN  ...                  NaN   
5                         12030.0  ...              42260.0   
6                             NaN  ...                  NaN   
7                             NaN  ...                  NaN   
8                             NaN  ...        

In [ ]:
rev.shape

(34926, 57)

In [ ]:
rev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34926 entries, 0 to 34925
Data columns (total 57 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Nom géographique GMS                                           34926 non-null  object 
 1   Code géographique                                              34926 non-null  object 
 2   Libellé géographique                                           34926 non-null  object 
 3   [DISP] Nbre de ménages fiscaux                                 31322 non-null  float64
 4   [DISP] Nbre de personnes dans les ménages fiscaux              31322 non-null  float64
 5   [DISP] Nbre d'unités de consommation dans les ménages fiscaux  31322 non-null  float64
 6   [DISP] 1ᵉʳ quartile (€)                                        5343 non-null   float64
 7   [DISP] Médiane (€)                                        

Les types de données sont cohérentes des revenues sont cohérentes

In [ ]:
rev.duplicated().sum() #Vérifier s'il y a des doublons dans la dataframe

np.int64(0)

Le dataframe ne présente aucun doublon.

In [ ]:
rev.isna().sum().sort_values(ascending=False).head(10) #Trouver les valeurs manquantes de chaque colonnes

[DISP] 3ᵉ quartile (€)                                           29583
[DISP] 1ᵉʳ quartile (€)                                          29583
[DISP] 7ᵉ décile (€)                                             29583
[DISP] 6ᵉ décile (€)                                             29583
[DISP] 4ᵉ décile (€)                                             29583
[DISP]3ᵉ décile (€)                                              29583
[DISP] 2ᵉ décile (€)                                             29583
[DISP] 1ᵉʳ décile (€)                                            29583
[DISP] Écart interquartile (€)                                   29583
[DISP] 9ᵉ décile (€)                                             29583
[DISP] Rapport interdécile 9ᵉ décile/1ᵉʳ decile                  29583
[DISP] S80/20                                                    29583
[DISP] Iice de Gini                                              29583
[DISP] Part des revenus d’activité (%)                           29583
[DISP] dont part des salaires et traitements(%)                  29583
[DISP] dont part des iemnités de chômage (%)                     29583
[DISP] 8ᵉ décile (€)                                             29583
[DISP] dont part des revenus des activités non salariées (%)     29583
[DISP] Part des impôts (%)                                       29583
[DISP] dont part des prestations logement (%)                    29583
[DISP] dont part des minima sociaux (%)                          29583
[DISP] dont part des prestations familiales (%)                  29583
[DISP] Part de l'ensemble des prestations sociales (%)           29583
[DISP] Part des revenus du patrimoine et autres revenus (%)      29583
[DISP] Part des pensions, retraites et rentes (%)                29583
[DEC] S80/S20                                                    29579
[DEC] Iice de Gini                                               29579
[DEC] Part des revenus d’activité (%)                            29579
[DEC] dont part des salaires et traitements (%)                  29579
[DEC] 7ᵉ décile (€)                                              29579
[DEC] 8ᵉ décile (€)                                              29579
[DEC] 9ᵉ décile (€)                                              29579
[DEC] 1ᵉʳ quartile (€)                                           29579
[DEC] 1ᵉʳ décile (€)                                             29579
[DEC] Part des ménages fiscaux imposés (%)                       29579
[DEC] Écart interquartile (€)                                    29579
[DEC] 3ᵉ quartile (€)                                            29579
[DEC] 6ᵉ décile (€)                                              29579
[DEC] 4ᵉ décile (€)                                              29579
[DEC] 3ᵉ décile  (€)                                             29579
[DEC] 2ᵉ décile (€)                                              29579
[DEC] dont part des iemnités de chômage (%)                      29579
[DEC] dont part des revenus des activités non salariées (%)      29579
[DEC] Part des pensions, retraites et rentes (%)                 29579
[DEC] Part des autres revenus (%)                                29579
[DEC] Rapport interdécile D9/D1                                  29579
[DISP] Nbre de personnes dans les ménages fiscaux                 3604
[DISP] Nbre de ménages fiscaux                                    3604
[DISP] Nbre d'unités de consommation dans les ménages fiscaux     3604
[DISP] Médiane (€)                                                3604
[DEC] Médiane (€)                                                 3603
[DEC] Nbre d'unités de consommation dans les ménages fiscaux      3603
[DEC] Nbre de personnes dans les ménages fiscaux                  3603
[DEC] Nbre de ménages fiscaux                                     3603
Code géographique                                                    0
Libellé géographique                                                 0
Nom géographique GMS    

Il y'a une quantité significative de valeurs manquantes de communes avec des revenue non enregistré, ce qui va poser problème lors de l'analyse, mais nous allons maintenant se concentrer sur les valeurs existantes pour faire l'analyse notament la colonne du revenue médian

## Observations clés
- Le code INSEE est présent dans les deux datasets
- Le revenu médian est manquant pour certaines communes (secret statistique)
- Les ordres de grandeur sont cohérents
- Les datasets sont prêts pouur le nettoyage et l'analyse
